src/needle.ts

```diff
@@ -52,26 +60,47 @@ export function makeRequest(payload: Payload): Promise<{ success: boolean; respo
 
   const options: needle.NeedleOptions = {
     headers: payload.headers,
-    timeout: payload.timeout,
+    open_timeout: 0, // No timeout
+    response_timeout: payload.timeout,
+    read_timeout: payload.timeout,
     family: payload.family,
     json: true,
     compressed: true, // sets 'Accept-Encoding' to 'gzip, deflate, br'
     follow_max: 5, // follow up to five redirects
     rejectUnauthorized: !global.ignoreUnknownCA, // verify SSL certificate
-    open_timeout: 0,
     agent,
   };
 
   emitter.apiRequestLog(`=> HTTP ${method?.toUpperCase()} ${url} ${data ?? ''}`.slice(0, 399));
-
-  return needle(method, url, data, options)
-    .then(response => {
+  let attempts = payload.attempts || MAX_RETRY_ATTEMPTS;
+  let retryDelay = payload.retryDelay || REQUEST_RETRY_DELAY;
```

Comment:
`question`: since all of these are numbers, are you using `||` in order to take that into consideration? 
If not, I might suggest using `??` to avoid counting `0` as a valid value